gatsby-browser.js

```diff
@@ -23,12 +23,10 @@ const checkDomain = () => {
 
 const addScript = (settings) => {
     const script = document.createElement('script')
-    const { async, text, src, id } = settings
 
-    if (async) script.async = settings['async']
-    if (text) script.text = settings['text']
-    if (src) script.src = settings['src']
-    if (id) script.id = settings['id']
+    Object.keys(settings).forEach((key) => {
```

Comment:
@mustofa-binary I get your point but generally, the `for..in` loop is better to be avoided as much as possible. because `for..in` loops not only iterate over keys of the object but also iterate over the prototype chain that may create bugs if is not intended. but `Object.keys()` is only extract the own keys of the object and does not go through prototype chain
here is an example:

```js
const parent = {
  name: 'dad',
};

const son = Object.create(parent);

son.age = 20;

for (let key in son) {
  console.log(key); // prints both "age" and "name"
}

console.log(Object.keys(son)); // prints only ["age"]
```

Reply:
actually the for in seems to be more as expected based on your example, since we expect son to hv name and age as well inherit from parent. but, here we want settings object not to be altered at all cost, so yes, Object.keys will do the job, fair point, thank you @mahdiaryayi-fs